src/2dplane/ObstacleGrid.cpp

```diff
@@ -24,6 +26,33 @@ Vector2i ObstacleGrid::gridSquareForLocation(const Vector2f &loc) const {
                     loc.y() / height() * discretizedHeight());
 }
 
+float ObstacleGrid::nearestObstacle(const Vector2f &state) const {
+    float initial = 2;
+    float c = initial; // closest distance
+    int x = (int)(state.x() / 20);
+    int y = (int)(state.y() / 20);
+    int dlim = 0;
+    int ulim = 0;
+    int llim = 0;
+    int rlim = 0;
+    
+    for (int i = x - c; i < x + c && i >= 0 && i < discretizedWidth(); i++) {
```

Comment:
Can you put a comment here describing what's going on?  I can tell from reading through it that you're expanding the search "radius", but it's a little dense and could benefit from a quick comment above to describe it.